### PR TITLE
IframeGenerator

### PR DESCRIPTION
**Pull request by thiloSchlemmer**

### CODE REVIEW

**Review by taulinger — COMMENTED** [review]
When the map included in the generator component is panned or zoomed by the user, the HTML snipped has to be updated (see v3)

File: src/modules/iframe/components/iframeGenerator/IFrameGenerator.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+				return {
+					...model,
+					autoWidth: data
+				};
+			case Update_Preview_Url:
+				return { ...model, previewUrl: data };
+		}
+	}
+
+	onAfterRender() {
```
Comment (taulinger) [inline src/modules/iframe/components/iframeGenerator/IFrameGenerator.js:65]: Only as a hint: `onAfterRender` comes with a parameter `firstTime`

File: src/modules/iframe/components/iframeGenerator/IFrameGenerator.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+			previewUrl: null
+		});
+		const {
+			TranslationService: translationService,
+			ShareService: shareService,
+			EnvironmentService: environmentService
+		} = $injector.inject('TranslationService', 'ShareService', 'EnvironmentService');
+		this._translationService = translationService;
+		this._shareService = shareService;
+		this._environmentService = environmentService;
```
Comment (taulinger) [inline src/modules/iframe/components/iframeGenerator/IFrameGenerator.js:39]: `this._environmentService` is currently unused

File: src/modules/iframe/components/iframeGenerator/IFrameGenerator.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+const Range_Min = 100;
+const Range_Max = 2000;
+
+/**
+ * Component to preview the embedded version
+ *
+ * @author thiloSchlemmer
+ * @author alsturm
+ */
+export class IFrameGenerator extends MvuElement {
```
Comment (taulinger) [inline src/modules/iframe/components/iframeGenerator/IFrameGenerator.js:25]: We should think about a coherent notation: `IFrame...` or `Iframe...` (see `IframeStatePlugin`)

File: test/modules/iframe/components/IFrameGenerator.test.js [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+		},
+		parent: {
+			location: 'iframe'
+		}
+	};
+
+	const environmentServiceMock = {
+		getWindow: () => mockWindow,
+		isEmbedded: () => false
+	};
```
Comment (taulinger) [inline test/modules/iframe/components/IFrameGenerator.test.js:33]: Currently unnecessary

**Review by alsturm — DISMISSED** [review]

**Review by taulinger — COMMENTED** [review]
Excellent! (just some minor change requests or hints)

File: src/modules/iframe/components/iframeGenerator/IFrameGenerator.js (hunk 3 above)
Comment (thiloSchlemmer) [reply src/modules/iframe/components/iframeGenerator/IFrameGenerator.js:25]: ok, then let's stick with `Iframe...`

**Review by taulinger — APPROVED** [review]
Let's go!

**Pull request merged**